ClientApp/src/pages/Browse.js

```diff
@@ -33,7 +37,7 @@ const Browse = () => {
                                     nobis.
                                 </p>
                                 <div className="d-flex justify-content-center">
-                                    <div className="button button--primary">View Blog</div>
+                                    <div onClick={() => goToBlog(blog.id)} className="button button--primary">View Blog</div>
```

Comment:
Can you use NavLink from react-router-dom rather than history? Just surrond the button with a  NavLink and remove the default styles from the a tag if there is any within the component.